src/SqlBinding/SqlAsyncCollector.cs

```diff
@@ -379,8 +381,55 @@ public static TableInformation RetrieveTableInformation(SqlConnection sqlConnect
                     primaryKeyMatchingQuery += $" AND ExistingData.{primaryKey} = NewData.{primaryKey}";
                 }
 
+                // 2. Get Columns Definition
+                string columnDefinitionQuery = $@"
```

Comment:
what benefits does grabbing the column types offer? More performance? or is it just required to use the JSON functionality?

Reply:
It is required for using OPENJSON

Reply:
https://docs.microsoft.com/en-us/sql/relational-databases/json/use-openjson-with-an-explicit-schema-sql-server?view=sql-server-ver15